src/components/query-viewer/query-viewer.tsx

```diff
@@ -236,32 +255,93 @@ export class QueryViewer implements GxComponent {
   /**
    * Type of the QueryViewer: Table, PivotTable, Chart, Card
    */
-  @Prop() readonly type: QueryViewerOutputType;
+  @Prop({ mutable: true }) type: QueryViewerOutputType;
 
   /**
    * if true the x Axes intersect at zero
    */
-  @Prop() readonly xAxisIntersectionAtZero: boolean;
+  @Prop({ mutable: true }) xAxisIntersectionAtZero: boolean;
 
   /**
    * Labels for XAxis
    */
-  @Prop() readonly xAxisLabels: QueryViewerXAxisLabels =
+  @Prop({ mutable: true }) xAxisLabels: QueryViewerXAxisLabels =
     QueryViewerXAxisLabels.Horizontally;
 
   /**
    * X Axis title
    */
-  @Prop() readonly xAxisTitle: string;
+  @Prop({ mutable: true }) xAxisTitle: string;
 
   /**
    * Y Axis title
    */
-  @Prop() readonly yAxisTitle: string;
+  @Prop({ mutable: true }) yAxisTitle: string;
+
+  /**
+   * If type == Map, this is the map type: Choropleth or Bubble
+   */
+  @Prop({ mutable: true }) mapType: QueryViewerMapType;
+
+  /**
+   * If type == Map, this is the region to display in the map
+   */
+  @Prop({ mutable: true }) region: QueryViewerRegion;
+
+  /**
+   * If type == Map and region = Continent, this is the continent to display in the map
+   */
+  @Prop({ mutable: true }) continent: QueryViewerContinent;
+
+  /**
+   * If type == Map and region = Country, this is the country to display in the map
+   */
+  @Prop({ mutable: true }) country: QueryViewerCountry;
 
   @Listen("queryViewerServiceResponse")
   handleServiceResponse(event: CustomEvent<QueryViewerServiceResponse>) {
     this.serviceResponse = event.detail;
+    this.setQueryViewerProperties(event.detail.Properties);
+  }
+
+  /**
+   * Set QueryViewer properties with values from the query obtained from the server (unless explicitly set in the web component)
+   */
+  private setQueryViewerProperties(properties: QueryViewerServiceProperties) {
+    if (properties) {
```

Comment:
I think this way of setting the properties does not work. If I set `show-values="false"` in my `gx-query-viewer`, the `properties.ShowValues` value will be used instead.

If we want to prioritize the gx-query-viewer's properties over `serviceResponse`, we must remove the default values of the properties in the gx-query-viewer interface and use the `??` operator instead of the `||` operator.
Also, we should check that this change won't break the Query Viewer when used with the Angular Gnerator in GeneXus.

When the defaults values are finally removed, this way of setting properties should work:
```js
this.queryTitle ??= properties.QueryTitle;
this.showValues ??= properties.ShowValues;
```